mmengine/visualization/visualizer.py

```diff
@@ -828,22 +1030,54 @@ def draw_polygons(
                 warnings.warn(
                     'Warning: The polygon is out of bounds,'
                     ' the drawn polygon may not be in the image', UserWarning)
-        if isinstance(line_widths, (int, float)):
-            line_widths = [line_widths] * len(polygons)
-        line_widths = [
-            min(max(linewidth, 1), self._default_font_size / 4)
-            for linewidth in line_widths
-        ]
-        polygon_collection = PolyCollection(
-            polygons,
-            alpha=alpha,
-            facecolor=face_colors,
-            linestyles=line_styles,
-            edgecolors=edge_colors,
-            linewidths=line_widths)
-
-        self.ax_save.add_collection(polygon_collection)
-        return self
+        if backend == 'matplotlib':
+            from matplotlib.collections import PolyCollection
+            edge_colors = color_val_matplotlib(edge_colors)  # type: ignore
+            face_colors = color_val_matplotlib(face_colors)  # type: ignore
+            if isinstance(line_widths, (int, float)):
+                line_widths = [line_widths] * len(polygons)
+            line_widths = [
+                min(max(linewidth, 1), self._default_font_size / 4)
+                for linewidth in line_widths
+            ]
+            polygon_collection = PolyCollection(
+                polygons,
+                alpha=alpha,
+                facecolor=face_colors,
+                linestyles=line_styles,
+                edgecolors=edge_colors,
+                linewidths=line_widths)
+
+            self.ax_save.add_collection(polygon_collection)
+            return self
+
+        elif backend == 'cv2':
+            line_widths = [line_widths] if isinstance(line_widths,
+                                                      (int,
+                                                       float)) else line_widths
+            assert len(polygons) == len(
```

Comment:
```suggestion
            line_widths = [line_widths] * len(polygons) if isinstance(line_widths,
                                                      (int,
                                                       float)) else line_widths
```

If the `if-else` statement is too long to be written in one line, we should write it separately.